File: LayoutTests/http/tests/cache/resources/xhr-vary-header-response.php
<?php
header("Vary: Origin");
header("Cache-Control: max-age=31536000");
if (isset($_SERVER['HTTP_ORIGIN'])) {
    header("Access-Control-Allow-Origin: http://127.0.0.1:8000");
    echo "Cross origin ";
} else {
    echo "Same origin ";
}
?>
response

File: LayoutTests/http/tests/cache/resources/xhr-vary-header-response.py
#!/usr/bin/env python3

import os
import sys

origin = True if os.environ.get('HTTP_ORIGIN', None) is not None else False

sys.stdout.write(
    'Vary: Origin\r\n'
    'Cache-Control: max-age=31536000\r\n'
    'Content-Type: text/html\r\n'
)

if origin:
    sys.stdout.write(
        'Access-Control-Allow-Origin: http://127.0.0.1:8000\r\n\r\n'
        'Cross origin '
    )
else:
    sys.stdout.write('\r\nSame origin ')

sys.stdout.write('response')